### PR TITLE
[KARAF-6897] Include conditional bundles on karaf-maven-plugin genera…

### PR DESCRIPTION
**Pull request by paulojramos**
…ted kar (offline deploy support)

### CODE REVIEW

File: tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/KarMojo.java [hunk 1]
```diff
@@ -209,6 +210,13 @@ private File resolveFile(String file) {
                         resources.add(resourceToArtifact(bundle.getLocation(), false));
                     }
                 }
+                for (Conditional conditional : feature.getConditional()) {
+                    for (BundleInfo bundle : conditional.getBundles()) {
+                        if (ignoreDependencyFlag || (!ignoreDependencyFlag && !bundle.isDependency())) {
```
Comment (skitt) [inline tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/KarMojo.java:215]:
```suggestion
                        if (ignoreDependencyFlag || !bundle.isDependency()) {
```

If the RHS of the `||` is evaluated at all, `!ignoreDependencyFlag` is always `true`.

**Review by skitt — CHANGES_REQUESTED** [review]
This might be a lot to ask, but would you consider adding an integration test to check this?

File: tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/KarMojo.java (hunk 1 above)
Comment (paulojramos) [reply tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/KarMojo.java]: I have simplified the conditions.
Wouldn't really be comfortable adding integration tests, being this my first contribution to the project.

**Comment by jbonofre** [comment]
@paulojramos thanks ! Can you please squash the commits in a single one ?

**Comment by paulojramos** [comment]
@jbonofre Done.

**Review by skitt — APPROVED** [review]

**Review by jbonofre — APPROVED** [review]

**Pull request merged**